### ISSUE TITLE
New Project creates two directories. 

### ISSUE THREAD

**Issue by gcasella** [GNS3/gns3-server#336]
Hey,
NOTE: Screenshots have been posted on the GNS3 jungle bug report (https://community.gns3.com/ideas/1898 -- please view this link for the screenshots)

Didn't know how else to name this bug title or where to post this on github (under gui or server).

What I'm experiencing with Version 1.3.10 is the following -- this also occurs in version 1.3.11 and 1.4.0 beta4;

Note: I use the GNS3 server on a remote server.
 

- Open GNS3 on Windows 10 and create new project right away (Test123).
- Create topology with QEMU Devices;
- Logs from GNS3 Server, appears it never logs that project Test123 has been created;

 
In previous version and using this in my old setup all files would beheld under one directory, but now I have two directories on both my remote server and local machine;

Local Directory: projects/Test123/
Remote Directory: projects/88f10b57-8350-4747-a191-7395722e5de3

 
It appears back inversion 1.2.3 it was working as i expected, you can see from an example shown below it would create my project directory and all sub-directories respectively;

**Comment by noplay** [comment]
Hi,

You mean the directory name is different on remote and local server right ? And  you can't use a system like Dropbox or a shared network directory.  Is it correct?

**Comment by gcasella** [comment]
Well that is how I've had this set up for quite some time. 
I have my own cloud solution using a product called "ownCloud" similar to dropbox, it run's on all of my systems at home to sync all data. My directory structure is as follows;

Local: C:\Users\gianluca.casella\ownCloud\GNS3
Remote: /home/gcasella/ownCloud/GNS3/
              - /projects
              - /images

these two directories will always synchronize with each other. 
But when I create a new project for example named "Test 1234" it will add it to my local sync folder which in turn goes to my remote server, but when i start adding devices the GNS3 Remote server just creates a new project directory using the UUID generator to create a new project directory. 

If you look at the link I posted in here that points to the jungle you will see the step by step that i took. And the outputs from both ends of my testing

**Comment by noplay** [comment]
It's not a "bug". The different path is normal since 1.3. Due to the fact multiple user can use the same server we can't use the same directory. We were doing that in the past but this has create problems. With people overwriting each other projects, or the same people with two projects with the same name on their computer.

Also we know that using a  share folder could sometimes corrupt the files. We sure VirtualBox and Google Drive could corrupt files.

But we understand your issue and we hope to provide a good solution in a futur version. With a correct way to have all the project files sync to the PC where the GUI is running.

**Comment by noplay** [comment]
We are sorry to disturb your workflow. It was a hard decision :(

**Comment by gcasella** [comment]
Hey Julien,
That's no problem at all. It was primarily to get some more background on it especially since it had done it in the past. 
It's just hard to easily find which project folder is associated with the local copy. Would it be possible to create the UUID project directory inside the project name directory instead of keeping them separate? 
For example,
Project Directory: /home/gns3/projects/test-project/uuid-dir/
That way everything regarding one major project is all under one directory.
I can be more descriptive in this example if you'd like
On 23 Oct 2015 10:46 am, Julien Duponchelle <notifications@github.com> wrote:We are sorry to disturb your workflow. It was a hard decision :(

**Comment by noplay** [comment]
It's an idea.

**Comment by gcasella** [comment]
Sounds good.

What I'll do in the meantime is the following;

 Whenever I create a new project and need to find any of the referencing project folders I will just open up the .gns3 file and see where it references. 

Thanks for the info you provided. Much appreciated.

Date: Fri, 23 Oct 2015 08:38:33 -0700

**Comment by noplay** [comment]
You can right click on a node and get the full path when clicking on show
in file browser (perhaps only with 1.4)

**Comment by noplay** [comment]
After more reflexion I'm concern by two things.

The file organization will be worse  and more confusing for all users if we add the project uuid in the project directory. 

And we know that file sync should be controlled by GNS3 not an external system because otherwise you have an high risk of data loss (Google Drive for example corrupt GNS3 project, NFS crash IOU).

**Comment by gcasella** [comment]
I haven't had any of those experiences with Corruption of a GNS3 project for a crash of IOU when running my solution.

I do understand where you are coming from with your concerns :).

This isn't a big issue for me anymore as I was able to figure out how I could make it work best for me. 

I did notice that the .gns3 file holds the Project UUID directory that is referenced on the remote server. 

So with my File Sync solution ownCloud (Not Google Drive/DropBox or NFS) I'm only sync'ing my local projects directory which hold the main .gns3 file and leave my project uuid folders where they should be. At this point I don't synchronize my gns3 remote servers with my local system anymore

There is one benefit I see to holding the project uuid inside of the project directory, its similar to how GNS3 had it in the past with I believe back in version 1.2.x where the <project-name>-files directory where it would hold all the contents that are stored in the Project UUID directory. However I know things have changed of course. I just wanted to give my two cents that's all :).
